src/libflate/lib.rs

```diff
@@ -35,13 +34,27 @@
 extern crate libc;
 
 use libc::{c_void, size_t, c_int};
+use std::fmt;
 use std::ops::Deref;
 use std::ptr::Unique;
 use std::slice;
 
+#[derive(Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct Error(());
+
+impl fmt::Debug for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        "decompression error".fmt(f)
+    }
+}
+
+impl Drop for Error {
```

Comment:
@nikomatsakis I can't think of a better solution than implementing `Drop`. Making it a struct means it can't be changed to an enum later on, and vice versa. The tuple struct is not special in that regard. The issue you linked to talks about tuple structs with all-pub fields, this one only has a private field.